### ISSUE TITLE
[Paper Note] Fantastic Pretraining Optimizers and Where to Find Them II: Hyperball Optimization, Kaiyue Wen+, arXiv'26, 2026.06

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#6318]
# URL
- https://arxiv.org/abs/2606.16899
# Authors
  - Kaiyue Wen
  - Xingyu Dang
  - Kaifeng Lyu
  - Tengyu Ma
  - Percy Liang
# Abstract
  - Matrix based optimizers such as Muon can substantially speed up language model pretraining, but their gains over AdamW are observed to shrink as model size and data scale grow when using standard constant decoupled weight decay. We propose Hyperball, a simple optimizer wrapper that addresses this issue. Given a base optimizer such as Adam or Muon, Hyperball sets the Frobenius norms of weight matrices and their corresponding optimizer updates to fixed constants. On Qwen3 style models up to 1.2B parameters, Muon Hyperball achieves 20--30% token equivalent speedup over weight decay baselines. Hyperball also improves learning rate transfer across widths and depths compared to decoupled weight decay. This method is motivated by prior theory showing that training with weight decay leads to an equilibrium weight norm that only depends on the training hyperparameters. Through this mechanism, the weight decay then decides the angular learning rate, i.e. how fast the direction of the weight matrix changes.
# Translation (by gpt-5-nano)
- Muon のような行列ベースのオプティマイザは言語モデルの事前学習を大幅に高速化することができるが、標準的な一定のデカップリングウェイトデケイを用いると、モデルサイズとデータ規模の拡大に伴い AdamW との差分の利得が縮小することが観察されている。私たちはこの問題に対処する単純なオプティマイザラッパーである Hyperball を提案する。基盤オプティマイザとして Adam や Muon を想定し、Hyperball は重み行列とそれに対応するオプティマイザ更新のフロベニウスノルムを固定定数に設定する。Qwen3 スタイルのモデル（最大約12億パラメータ）において、Muon Hyperball はウェイトデケイをベースラインとした場合、20〜30% のトークン換算の速度向上を達成する。Hyperball はまた、デカップリングウェイトデケイと比較して、幅と深さを超えた学習率の伝達を改善する。この手法は、ウェイトデケイを用いた訓練が訓練のハイパーパラメータのみに依存する平衡状態のウェイトノルムをもたらすという既存理論に動機づけられている。これを通じて、ウェイトデケイは角度学習率、すなわちウェイト行列の方向がどれだけ速く変化するかを決定する。
# Summary (by gpt-4o-mini)
- Hyperballは、行列ベースのオプティマイザMuonに対するシンプルなラッパーで、固定定数に重み行列のフロベニウスノルムを設定することで、言語モデルの事前学習を高速化する。Qwen3スタイルのモデルでは、Muon Hyperballが20〜30%の速度向上を実現し、デカップリングウェイトデケイと比べて学習率の伝達を改善する。これにより、訓練中のウェイトデケイがハイパーパラメータに依存する状態を促進する。

**Comment by AkihikoWatanabe** [comment]
元ポスト:https://x.com/askalphaxiv/status/2087221609312436717?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
関連:
- #4261
